DXMainClient/DXGUI/Multiplayer/CnCNet/ChoiceNotificationBox.cs

```diff
@@ -0,0 +1,189 @@
+            lblHeader.AnchorPoint = new Vector2(ClientRectangle.Width / 2, 12);
+            lblHeader.TextAnchor = LabelTextAnchorInfo.CENTER;
+            lblHeader.Text = "MAKE A CHOICE";
+            AddChild(lblHeader);
+
+            gameIconPanel = new XNAPanel(WindowManager);
+            gameIconPanel.Name = "gameIconPanel";
+            gameIconPanel.ClientRectangle = new Rectangle(12, lblHeader.Bottom + 6, 16, 16);
+            gameIconPanel.DrawBorders = false;
+            gameIconPanel.BackgroundTexture = AssetLoader.TextureFromImage(ClientCore.Properties.Resources.dtaicon);
+            AddChild(gameIconPanel);
+
+            lblSender = new XNALabel(WindowManager);
+            lblSender.Name = "lblSender";
+            lblSender.FontIndex = 1;
+            lblSender.ClientRectangle = new Rectangle(gameIconPanel.Right + 3, lblHeader.Bottom + 6, 0, 0);
+            lblSender.Text = "fonger";
+            AddChild(lblSender);
+
+            lblChoiceText = new XNALabel(WindowManager);
+            lblChoiceText.Name = "lblChoiceText";
+            lblChoiceText.FontIndex = 1;
+            lblChoiceText.ClientRectangle = new Rectangle(12, lblSender.Bottom + 6, 0, 0);
+            lblChoiceText.Text = "What do you want to do?";
+            AddChild(lblChoiceText);
+
+            affirmativeButton = new XNAButton(WindowManager);
+            affirmativeButton.FontIndex = 1;
+            affirmativeButton.ClientRectangle = new Rectangle(ClientRectangle.Left + 8, lblChoiceText.Bottom + 6, 75, 23);
+            affirmativeButton.IdleTexture = AssetLoader.LoadTexture("75pxbtn.png");
+            affirmativeButton.HoverTexture = AssetLoader.LoadTexture("75pxbtn_c.png");
+            affirmativeButton.HoverSoundEffect = new EnhancedSoundEffect("button.wav");
+            affirmativeButton.Name = "affirmativeButton";
+            affirmativeButton.Text = "Yes";
+            affirmativeButton.LeftClick += AffirmativeButton_LeftClick;
+            AddChild(affirmativeButton);
+
+            negativeButton = new XNAButton(WindowManager);
+            negativeButton.FontIndex = 1;
+            negativeButton.ClientRectangle = new Rectangle(ClientRectangle.Width - (75 + 8), lblChoiceText.Bottom + 6, 75, 23);
+            negativeButton.IdleTexture = AssetLoader.LoadTexture("75pxbtn.png");
+            negativeButton.HoverTexture = AssetLoader.LoadTexture("75pxbtn_c.png");
+            negativeButton.HoverSoundEffect = new EnhancedSoundEffect("button.wav");
+            negativeButton.Name = "negativeButton";
+            negativeButton.Text = "No";
+            negativeButton.LeftClick += NegativeButton_LeftClick;
+            AddChild(negativeButton);
+
+            base.Initialize();
+        }
+
+        public void Show(string headerText, Texture2D gameIcon, string sender, string choiceText, string affirmativeText, string negativeText, int timeout)
+        {
+            Visible = true;
+            Enabled = true;
```

Comment:
Those two lines can and should be replaced with call of `Enable()` method for this object.